enhancements/custom-alert-configuration.md

```diff
@@ -0,0 +1,439 @@
+max by (namespace, poddisruptionbudget) (kube_poddisruptionbudget_status_current_healthy >= kube_poddisruptionbudget_status_desired_healthy and on (namespace, poddisruptionbudget) (kube_poddisruptionbudget_status_expected_pods > 0) unless on(namespace, poddisruptionbudget) (ocp_kubevirt_vmi_owned_pdbs))
+```
+
+Kubevirt would maintain/own `ocp_kubevirt_vmi_owned_pdbs` metric and ensure
+only the valid PDBs are ignored. This should prevent users from inserting their
+own PDBs into this ignore list.
+
+The disadvantage is that it is not scalable for more components. The metric
+describes only KubeVirt use case.
+
+### Silencing the alert
+
+It is possible to partially silence alerts that are firing either manually via
+an [openshift console][7] or via an [alertmanager API][8].
+
+It can filter out alerts according to known prometheus labels. Here we check
+for the poddisruptionbudget label with a `kubevirt-disruption-budget-.*`  value.
+
+```json
+curl -s -X POST -H "Authorization: Bearer ${TOKEN}" -H 'Accept: application/json' -H 'Content-Type: application/json' "${ALERTMANANER_URL}/api/v2/silences" -d '{
+	"status": {
+		"state": "active"
+},
+	"comment": "kubevirt PDBs",
+	"createdBy": "kube:admin",
+	"startsAt": "2023-01-12T20:37:30.995Z"
+	"endsAt": "2024-01-12T22:32:07.000Z",
+	"matchers": [
+  	{
+    	"isRegex": false,
+        "isEqual": true,
+    	"name": "alertname",
+    	"value": "PodDisruptionBudgetAtLimit"
+  	},
+  	{
+    	"isEqual": true,
+    	"isRegex": true,
+    	"name": "poddisruptionbudget",
+    	"value": "kubevirt-disruption-budget-.*"
+  	},
+  	{
+    	"isEqual": true,
+    	"isRegex": false,
+    	"name": "severity",
+    	"value": "warning"
+  	}
+	],
+  },
+'
+```
+
+This would work in a sense that it would stop notifications for this alert and
+would be silenced in the UI.
+
+There might be following problems with this approach:
+- In the UI it shows the alert is firing and being silenced at the same time.
+  It might prompt users to action when seeing the alert is silenced. When you
+  discover the alert via an API it just shows this alert is firing, and you
+  need to match this to silences to get the whole picture.
+- The silence is meant as a temporary solution and there needs to be `endsAt`
+  date and time defined. It is also possible to expire the silence with one
+  click in the UI. We would need to manage the silence to be sure it is always
+  present and not expired.
+- As mentioned above this is not a standard solution, and we are not shipping
+  any silences with openshift AFAIK. We would probably need a controller that
+  would reconcile the alert through alertmanager API.
+- It is not precise enough. That is we would be filtering according to the
+  known prometheus labels, which is a poddisruptionbudget name in this case.
+  Users could hijack this functionality and create their own PDBs in their
+  namespace with a kubevirt-disruption-budget prefix that would get captured
+  by this silence.
```

Comment:
You could also filter by `namespace`, and forging `namespace` on `kube_poddisruptionbudget_status_current_healthy` is more difficult than just naming a PDB.  Although it's also unclear to me [what the guards on metrics impersonation are][1], so it might not be _much_ more difficult.

[1]: https://github.com/openshift/enhancements/pull/1328#discussion_r1091519847

Reply:
The problem is that kubevirt PDBs can occur in any namespace as they are guarding user virtual machines